annotation-file-utilities/src/annotator/find/InheritedSymbolFinder.java

```diff
@@ -0,0 +1,211 @@
+import javax.lang.model.util.Elements;
+import javax.lang.model.util.Types;
+
+import annotations.util.JVMNames;
+import annotator.Source;
+
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.LinkedHashMultimap;
+import com.google.common.collect.Multimap;
+import com.google.common.collect.Multimaps;
+
+import com.sun.tools.javac.code.Symbol;
+import com.sun.tools.javac.model.JavacElements;
+
+/**
+ * Provides static access to JVML descriptors of methods for which
+ *  implementations are inherited in a given class.
+ *
+ * @author dbro
+ */
+public class InheritedSymbolFinder {
+  private static Source source = null;
+
+  // class (flat) name -> method index
+  private static Map<String, Multimap<String, ExecutableElement>> cache =
+      new TreeMap<String, Multimap<String, ExecutableElement>>();
+
+  // keys: class (flat) name
+  // values: multimap of method (simple) name to matching method elems
+  private static Map<String, Multimap<String, ExecutableElement>> cash =
+      new HashMap<String, Multimap<String, ExecutableElement>>();
+
+  // initialized once and for all
+  private static Multimap<String, ExecutableElement> objectMethods = null;
+
+  private InheritedSymbolFinder() {}
+
+  // Needed for obtaining instances of utility classes Types and Elements.
+  // TODO: make methods non-static and pass source to constructor instead.
+  public static void setSource(Source source) {
+    if (InheritedSymbolFinder.source != source) {
+      InheritedSymbolFinder.source = source;
+      cache.clear();
+      cash.clear();
+      if (source != null) {
+        // initialize objectMethods
+        if (objectMethods == null) {
+          TypeElement o =
+              source.getElementUtils().getTypeElement("java.lang.Object");
+          objectMethods =
+              LinkedHashMultimap.<String, ExecutableElement>create();
+          for (Element e : o.getEnclosedElements()) {
+            if (e.getKind() == ElementKind.METHOD) {
+              String key = keyFor((Symbol.MethodSymbol) e);
+              objectMethods.put(key, (ExecutableElement) e);
+            }
+          }
+        }
+        cash.put("java.lang.Object", objectMethods);
+      }
+    }
+  }
+
+  /**
+   * Finds whether a method matching a JVML descriptor has an inherited
+   *  implementation in a given class.
+   *
+   * @param classSymbol symbol table entry for a class
+   * @param methodDescriptor JVML descriptor for a method
+   * @return whether a matching method exists in the class
+   */
+  public static boolean isInheritedIn(Symbol.ClassSymbol classSymbol,
+      String methodDescriptor) {
+    Multimap<String, ExecutableElement> inherited = getInherited(classSymbol);
+    int ix = methodDescriptor.indexOf('(');
+    String key = methodDescriptor.substring(0, ix);
+    Collection<ExecutableElement> elems = inherited.get(key);
+
+    // There may be multiple methods stored under the same key (bare
```

Comment:
Please use standard terminology rather than making up new terms.  By "bare method name" do you mean "simple method name"?